src/rules/if_change_then_change.rs

```diff
@@ -99,47 +149,88 @@ pub fn ictc(
+                        .get(&block.path)
+                        .unwrap_or(&HashSet::new()),
+                ) {
+                    blocks.push(block);
+                }
+            }
         }
     }
 
     let blocks_by_path: HashMap<&PathBuf, &IctcBlock> =
         blocks.iter().map(|b| (&b.path, b)).collect();
 
-    let ret: Vec<diagnostic::Diagnostic> = blocks
-        .iter()
-        .filter(|b| blocks_by_path.get(&b.thenchange).is_none())
-        .map(|b| diagnostic::Diagnostic {
-            range: diagnostic::Range {
-                path: b.path.to_str().unwrap().to_string(),
-                start: diagnostic::Position {
-                    line: b.begin as u64,
-                    character: 0,
-                },
-                end: diagnostic::Position {
-                    line: b.end as u64,
-                    character: 0,
-                },
-            },
-            severity: diagnostic::Severity::Error,
-            code: "if-change-then-change".to_string(),
-            message: format!(
-                "Expected change in {} because {} was modified",
-                b.thenchange.display(),
-                b.path.display(),
-            ),
-        })
-        .collect();
+    let mut diagnostics: Vec<diagnostic::Diagnostic> = Vec::new();
+
+    for block in &blocks {
+        match &block.thenchange {
+            ThenChange::RemoteFile(remote_file) => {
+                // Handle the case where `thenchange` is a `RemoteFile`
+                println!("Remote file: {}", remote_file);
+            }
+            ThenChange::RepoFile(local_file) => {
+                // Check if the repo file exists - if it was deleted this is a warning
+                if !Path::new(local_file).exists() {
+                    diagnostics.push(diagnostic::Diagnostic {
+                        range: block.get_range(),
+                        severity: diagnostic::Severity::Error,
+                        code: "if-change-file-does-not-exist".to_string(),
+                        message: format!("ThenChange {} does not exist", local_file.display(),),
+                    });
+                }
+                // If target file was not changed raise issue
+                if blocks_by_path.get(&local_file).is_none() {
+                    diagnostics.push(diagnostic::Diagnostic {
+                        range: block.get_range(),
+                        severity: diagnostic::Severity::Error,
+                        code: "if-change-then-change-this".to_string(),
+                        message: format!(
+                            "Expected change in {} because {} was modified",
+                            local_file.display(),
+                            block.path.display(),
+                        ),
+                    });
+                }
+            }
+
+            ThenChange::None => {
+                // Handle the case where `thenchange` is `None`
+                println!("No change");
```

Comment:
This should be an error, no? If there's no `thenchange` there's no point to having an `ifchange`, right?

Reply:
Cleaned this up. It was really a case of Option and didn't need a None condition